// node-json-db/dist/lib/JsonDBConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigWithAdapter = exports.Config = void 0;
const path = require("path");
const JsonAdapter_1 = require("../adapter/data/JsonAdapter");
const AtomicFileAdapter_1 = require("../adapter/file/AtomicFileAdapter");
class Config {
    adapter;
    filename;
    saveOnPush;
    separator;
    constructor(filename, saveOnPush = true, humanReadable = false, separator = '/', syncOnSave = false) {
        this.filename = filename;
        // Force json if no extension
        if (path.extname(filename) === "") {
            this.filename += ".json";
        }
        this.saveOnPush = saveOnPush;
        this.separator = separator;
        this.adapter = new JsonAdapter_1.JsonAdapter(new AtomicFileAdapter_1.AtomicFileAdapter(this.filename, syncOnSave), humanReadable);
    }
}
exports.Config = Config;
class ConfigWithAdapter {
    adapter;
    saveOnPush;
    separator;
    constructor(adapter, saveOnPush = true, separator = '/') {
        this.adapter = adapter;
        this.saveOnPush = saveOnPush;
        this.separator = separator;
    }
}
exports.ConfigWithAdapter = ConfigWithAdapter;
